src/include/duckdb/common/types/string_type.hpp

```diff
@@ -33,10 +33,15 @@ struct string_t {
 			// zero initialize the prefix first
 			// this makes sure that strings with length smaller than 4 still have an equal prefix
 			memset(prefix, 0, PREFIX_LENGTH);
+			value_.data = nullptr;
```

Comment:
Is this assignment necessary? I believe the memcpy afterwards will override the data in the union again. It seems like this is superfluous.

Reply:
depends on the length of the string no? if length = 5 then there is uninitialised stuff in .data

Reply:
That is true, but it shouldn't matter no? If length =5 then IsInlined() is true, and the pointer should never be read.

Just wondering because this is done for every string value and any extra ops there could potentially decrease performance.

Reply:
Well if its certainly never read then indeed it does not matter whether its initialised.